labs/content/labs/05-forms.md

```diff
@@ -180,48 +180,26 @@ initialValues={ {
 
 * Now let's actually use the form we just built.
 * Open **/src/employees/EmployeeDetail.js**
-* Let's setup our propTypes, first. We already have the `history` prop, but we need to include an `employee` prop as well, like so:
+* Let's setup our local state, first. We will need to set the initial state of our employee object, like so:
 
 ```jsx
-EmployeeDetail.propTypes = {
-  employee: PropTypes.object.isRequired,
-  history: PropTypes.object,
-};
-
-EmployeeDetail.defaultProps = {
-  employee: {}
-};
-```
-
-* Next, let's hook up to Redux. The only thing we'll need from the Redux state is the selected employee, like so:
-
-```jsx
-const mapStateToProps = (state, props) => {
-  const { match } = props;
-  const { _id } = match.params;
-  return {
-    employee: state.employees.data.find(employee => employee._id === _id)
+  state = {
+    employee: null
   };
-};
 ```
 
-* We use react-router to give us the parameters that `match`-ed from the `Route` - in this instance, we'll get the employee ID from the URL. We use that to find the corresponding employee record. If none exists, then we'll get `undefined`
-
-* This component will also need the ability to update the Redux state through using the `EmployeeActions`, like so:
+* Next, let's fetch the employee data. We'll want to call the API when the component mounts and set the local state with the employee object. We can use the `axios` library to help us make HTTP requests.
 
 ```jsx
-const mapDispatchToProps = {
-  onCreate: EmployeeActionCreators.createEmployee,
-  onUpdate: EmployeeActionCreators.updateEmployee,
-  getEmployee: EmployeeActionCreators.getEmployee
-};
+  async componentDidMount() {
+    const { match } = this.props;
+    const { _id } = match.params;
+    const { data: employee } = await Axios.get(url(_id));
```

Comment:
Where is this url function coming from?

🤔

Reply:
First of all, whatcha doin in here? 😂 Second, the function is already in the file that they are editing.. Might be a bit confusing though?

Reply:
I've found that ~75% of people just copy and paste--so if it makes sense in context that's probably OK!

And 🤷‍♂️